packages/react-components/src/components/TypingIndicator.tsx

```diff
@@ -56,57 +52,85 @@ const getDefaultComponents = (
     const additionalCharCount =
       2 * (countOfUsersMentioned - 1) + (typingUser.displayName ? typingUser.displayName.length : 0);
     if (totalCharacterCount + additionalCharCount <= MAXIMUM_LENGTH_OF_TYPING_USERS || countOfUsersMentioned === 1) {
-      typingUsersMentioned.push(typingUser);
+      if (typingUser.displayName) {
+        typingUsersMentioned.push(typingUser.displayName);
+      }
       totalCharacterCount += additionalCharCount;
       countOfUsersMentioned += 1;
     } else {
       break;
     }
   }
 
-  typingUsersMentioned.forEach((typingUser, index) => {
-    displayComponents.push(
-      <span
-        className={mergeStyles(typingIndicatorListStyle, styles?.typingUserDisplayName)}
-        key={`typing indicator display string ${index}`}
-      >
-        {index < typingUsers.length - 1 ? typingUser.displayName + ', ' : typingUser.displayName}
-      </span>
-    );
-  });
-
   const countOfUsersNotMentioned = typingUsers.length - typingUsersMentioned.length;
-  if (countOfUsersNotMentioned > 0) {
-    displayComponents.push(
-      <span className={mergeStyles(typingIndicatorVerbStyle, styles?.typingString)} key={'UsersCountKey'}>
-        {` and ${countOfUsersNotMentioned} other${countOfUsersNotMentioned === 1 ? '' : 's'}`}
-      </span>
-    );
+
+  let typingString = '';
```

Comment:
The `getIndicatorComponents` function is really long, consider pulling this nice if statement out into a unit tested utility function